### PR TITLE
Fix: dont break on recursive json schemas

### PR DESCRIPTION
**Pull request by jeremyschiemann**
Hi, 

this is the fix i mentioned in #435 
I limited the recursion limit to a depth of 100 which should be enough.

### CODE REVIEW

File: src/schemathesis/schemas.py [hunk 1]
```diff
@@ -291,18 +291,20 @@ def resolve(self, item: List) -> List:  # pylint: disable=function-redefined
         pass
 
     # pylint: disable=function-redefined
-    def resolve(self, item: Union[Dict[str, Any], List]) -> Union[Dict[str, Any], List]:
+    def resolve(self, item: Union[Dict[str, Any], List], recursion_level=0) -> Union[Dict[str, Any], List]:
```
Comment (Stranger6667) [inline src/schemathesis/schemas.py]: please, add a type annotation to `recursion_level` : `recursion_level: int = 0`

**Comment by jeremyschiemann** [comment]
I added the annotation, changelog entry and ran the black command

**Comment by Stranger6667** [comment]
I completely forgot about two `overload` for this `resolve` function, could you, please update them as well?
Also, we use conventional commits format, please, take a look and adjust your commit messages - https://github.com/kiwicom/schemathesis/pull/441/checks?check_run_id=530315178

Sorry for inconvenience

**Comment by jeremyschiemann** [comment]
No worries, I hope everything is fine now.

I can't edit the older commit messages tho

**Comment by Stranger6667** [comment]
Great, I'll squash them then

**Pull request merged**